src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,2181 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2024 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <test/jtx/utility.h>
+#include <xrpl/protocol/Batch.h>
+#include <xrpl/protocol/Feature.h>
+#include <xrpl/protocol/STParsedJSON.h>
+#include <xrpl/protocol/Sign.h>
+#include <xrpl/protocol/TxFlags.h>
+#include <xrpl/protocol/jss.h>
+
+namespace ripple {
+namespace test {
+
+class Batch_test : public beast::unit_test::suite
+{
+    struct TestBatchData
+    {
+        std::string result;
+        std::string txHash;
+    };
+
+    struct TestSignData
+    {
+        int index;
+        jtx::Account account;
+    };
+
+    Json::Value
+    getTxByIndex(Json::Value jrr, std::uint8_t index)
+    {
+        for (auto const& txn : jrr[jss::result][jss::ledger][jss::transactions])
+        {
+            if (txn[jss::metaData][sfTransactionIndex.jsonName] == index)
+                return txn;
+        }
+        return {};
+    }
+
+    void
+    validateBatch(
+        jtx::Env& env,
+        TxID const& batchId,
+        std::vector<TestBatchData> const& batchResults)
+    {
+        Json::Value params;
+        params[jss::ledger_index] = env.current()->seq() - 1;
+        params[jss::transactions] = true;
+        params[jss::expand] = true;
+        auto const jrr = env.rpc("json", "ledger", to_string(params));
+        // std::cout << "jrr: " << jrr << std::endl;
+
+        // Validate the number of transactions in the ledger
+        auto const transactions =
+            jrr[jss::result][jss::ledger][jss::transactions];
+        BEAST_EXPECT(transactions.size() == batchResults.size() + 1);
+
+        // Validate ttBatch is correct index
+        auto const txn = getTxByIndex(jrr, 0);
```

Comment:
Is this assuming that Batch transactions are the only ones in the ledger?

Reply:
This is checking that the Batch outer is in whatever index I specify. In this case its hardcoded but I want to add a few more tests with txns infront of the Batch and in that case I wanted to make sure the batch was in the correct index instead of iterating through the ledger transactions.